sympy/matrices/expressions/matadd.py

```diff
@@ -34,7 +34,7 @@ class MatAdd(MatrixExpr, Add):
 
     identity = GenericZeroMatrix()
 
-    def __new__(cls, *args, evaluate=False, check=False, _sympify=True):
+    def __new__(cls, *args, evaluate=False, check=True, _sympify=True):
         if not args:
```

Comment:
I'm still thinking that maybe we should ignore the check argument and always check...

Reply:
I also think that is the correct way. Although it may be useful to bypass the checking in some situations...

Maybe set the default to None deprecate passing as value for one release?